hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/DecommissionStatusSubCommand.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hdds.scm.cli.datanode;
+
+import com.google.common.base.Strings;
+import org.apache.hadoop.hdds.cli.HddsVersionProvider;
+import org.apache.hadoop.hdds.protocol.DatanodeDetails;
+import org.apache.hadoop.hdds.protocol.proto.HddsProtos;
+import org.apache.hadoop.hdds.scm.cli.ScmSubcommand;
+import org.apache.hadoop.hdds.scm.client.ScmClient;
+import org.apache.hadoop.hdds.scm.container.ContainerID;
+import org.apache.hadoop.hdds.scm.container.ContainerInfo;
+import org.apache.hadoop.hdds.scm.container.ContainerNotFoundException;
+import org.apache.hadoop.hdds.scm.pipeline.Pipeline;
+import picocli.CommandLine;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static org.apache.hadoop.hdds.protocol.proto.HddsProtos.NodeOperationalState.DECOMMISSIONED;
+import static org.apache.hadoop.hdds.protocol.proto.HddsProtos.NodeOperationalState.DECOMMISSIONING;
+
+/**
+ * Handler to print decommissioning nodes status.
+ */
+@CommandLine.Command(
+    name = "decommissionStatus",
+    description = "Show decommission status for datanodes",
+    mixinStandardHelpOptions = true,
+    versionProvider = HddsVersionProvider.class)
+
+public class DecommissionStatusSubCommand extends ScmSubcommand {
+
+  @CommandLine.Option(names = { "--decommissioned" },
+      description = "Show info of decommissioned nodes too",
+      defaultValue = "false")
+  private boolean decommissioned;
+
+  @CommandLine.Option(names = { "--id" },
+      description = "Show info by datanode UUID",
+      defaultValue = "")
+  private String uuid;
+
+  @Override
+  public void execute(ScmClient scmClient) throws IOException {
+    List<HddsProtos.Node> decommissioningNodes;
+
+    if (!Strings.isNullOrEmpty(uuid)) {
+      decommissioningNodes = scmClient.queryNode(DECOMMISSIONING, null,
+          HddsProtos.QueryScope.CLUSTER, "").stream().filter(p ->
+          p.getNodeID().getUuid().equals(uuid)).collect(Collectors.toList());
```

Comment:
Since we're only concerned with one datanode in this case, we can avoid asking the SCM for all datanodes that are decommissioning. Instead, let's just ask for that one datanode so we don't have to transfer not needed data over the network.

Reply:
As this is not a trivial change, I'll handle this in [HDDS-9648](https://issues.apache.org/jira/browse/HDDS-9648) APIs to fetch only one DN information will be introduced under it